### PR TITLE
Peak1D and Peak2D: change intensity type to double as well

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

K I did some tests how switching the intensity type from float to double (like for m/z) affects OpenMS.
Surprisingly, hick-ups were rather small and only impacted a few tests.

Note: in MSSpectrum, we did not save memory with the float anyway due to alignment of Peak1D structs
as array of structs.

Some relevant questions are: 
- is this helpful? e.g. 
- simplifies switching to AoS?
- use Eigen Map capabilities to directly process peaks and intensities. Is it worth it or just cool?
- should we even go from double to float for both m/z and intensity? This would approx. half memory usage.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by cbielow** [comment]
not convinced this is the way to go.
Sure, the memory footprint is identical, due to padding in Peak1D and Peak2D.
However, doing non-trivial computations on doubles, like `log()` will take more time, since it takes longer to converge.
Trivial operations like addition should have the same speed.
Furthermore, if we ever move to SoA, then using `float` will allow twice the throughput for SIMD instructions.
However, using float for `m/z` may be harmful since there may not be enough precision for UltraHiRes data (in the future).
Bottom line: I would leave it as is: float for intensity, and double for m/z. You can probably make a good case for float in RT/IM.
